tests/chebyshev_test.py

```diff
@@ -0,0 +1,812 @@
+      jtu.format_shape_dtype_string((length1+trailing1,), dtype1),
+      trailing1,
+      jtu.format_shape_dtype_string((length2+trailing1,), dtype2),
+      trailing2),
+      "dtype1": dtype1, "length1": length1, "trailing1": trailing1,
+      "dtype2": dtype2, "length2": length2, "trailing2": trailing2}
+      for dtype1 in all_dtypes
+      for dtype2 in all_dtypes
+      for length1 in [1, 5, 10]
+      for length2 in [1, 5, 10]
+      for trailing1 in [0, 1]
+      for trailing2 in [0, 1]))
+  def testChebdiv(self, dtype1, dtype2, length1, length2, trailing1, trailing2):
+    rng = jtu.rand_default(self.rng())
+
+    def args_maker():
+      c1 = rng((length1,), dtype1)
+      c2 = rng((length2,), dtype2)
+
+      return (jnp.concatenate([c1, jnp.zeros(trailing1, c1.dtype)]),
+              jnp.concatenate([c2, jnp.zeros(trailing2, c2.dtype)]))
+
+    jnp_fn = partial(cs.chebdiv, trim=False)
+
+    with jtu.strict_promotion_if_dtypes_match([dtype1, dtype2]):
+      self._CheckAgainstNumpy(ncs.chebdiv, cs.chebdiv, args_maker, check_dtypes=False,
+                              tol=1e-6)
+      self._CompileAndCheck(jnp_fn, args_maker)
+
+  @parameterized.named_parameters(jtu.cases_from_list(
+    {"testcase_name": "_dtype={}_trailing={}_pow={}".format(
+      jtu.format_shape_dtype_string((length+trailing,), dtype),
+      trailing, pow),
+      "dtype": dtype, "length": length, "trailing": trailing, "pow": pow}
+      for dtype in all_dtypes
+      for length in [1, 5, 10]
+      for trailing in [0, 1]
+      for pow in [0,1,2,10]))
+  def testChebpow(self, dtype, length, trailing, pow):
+    rng = jtu.rand_default(self.rng())
+
+    def args_maker():
+      c = rng((length,), dtype)
+      return (jnp.concatenate([c, jnp.zeros(trailing, c.dtype)]), pow)
+
+    jnp_fn = lambda c, _p: partial(cs.chebpow, pow=pow, trim=False)(c)
+
+    self._CheckAgainstNumpy(ncs.chebpow, cs.chebpow, args_maker,
+                            check_dtypes=False, tol=1e-4)
+    self._CompileAndCheck(jnp_fn, args_maker)
+
+  @parameterized.named_parameters(jtu.cases_from_list(
+    {"testcase_name": "_dtype_c={}_trailing={}_dtype_x={}_tensor={}".format(
+      jtu.format_shape_dtype_string((length+trailing,), dtype_c),
+      trailing, jtu.format_shape_dtype_string((pts,), dtype_x), tensor),
+      "dtype_c": dtype_c, "length": length, "trailing": trailing,
+      "dtype_x": dtype_x, "pts": pts, "tensor": tensor}
+      for dtype_x in all_dtypes
+      for dtype_c in all_dtypes
+      for length in [1, 5, 10]
+      for trailing in [0, 1]
+      for pts in [0,1,2,5]
+      for tensor in [0, 1]))
+  def testChebval(self, dtype_x, dtype_c, length, trailing, pts, tensor):
+    rng = jtu.rand_default(self.rng())
+
+    def args_maker():
+      c = rng((length,), dtype_c)
+      if pts == 0:
+        p, = rng(1, dtype_x)
+      else:
+        p = rng((pts,), dtype_x)
+      t = tensor == 1
+      return (p, jnp.concatenate([c, jnp.zeros(trailing, c.dtype)]), t)
```

Comment:
This is concerning - the numpy version always gets `t = 1`, while the JAX version below ignores the value of `t` and uses `tensor`, which can be zero or 1. If this test is passing, it makes me think the implementation is incorrect.